recipe-server/normandy/recipes/tests/test_models.py

```diff
@@ -44,6 +45,50 @@ def test_recipes_used_by_empty(self):
         RecipeFactory.create_batch(2, action=action, enabled=False)
         assert list(action.recipes_used_by) == []
 
+    def test_validate_arguments_it_works(self):
+        action = ActionFactory(name='nothing special')
+        # does not raise an exception
+        action.validate_arguments({})
+
+    def test_validate_arguments_preference_exeriments_unique_branch_slugs(self):
+        action = ActionFactory(name='preference-experiment')
+        arguments = {
+            'slug': 'test',
+            'branches': [
+                {'slug': 'unique', 'value': 'a'},
+                {'slug': 'duplicate', 'value': 'b'},
+                {'slug': 'duplicate', 'value': 'c'}
+            ]
+        }
+        with pytest.raises(serializers.ValidationError) as exc_info:
+            action.validate_arguments(arguments)
+        error = action.errors['duplicate_branch_slug']
+        assert exc_info.value.detail == {'arguments': {'branches': {2: {'slug': error}}}}
+
+    def test_validate_arguments_preference_exeriments_unique_branch_values(self):
+        action = ActionFactory(name='preference-experiment')
+        arguments = {
+            'slug': 'test',
+            'branches': [
+                {'slug': 'a', 'value': 'unique'},
+                {'slug': 'b', 'value': 'duplicate'},
+                {'slug': 'c', 'value': 'duplicate'}
+            ]
+        }
+        with pytest.raises(serializers.ValidationError) as exc_info:
+            action.validate_arguments(arguments)
+        error = action.errors['duplicate_branch_value']
+        assert exc_info.value.detail == {'arguments': {'branches': {2: {'value': error}}}}
+
+    def test_validate_arguments_preference_experiments_unique_experiment_slug(self):
+        action = ActionFactory(name='preference-experiment')
+        arguments = {'slug': 'duplicate', 'branches': []}
+        RecipeFactory(action=action, arguments=arguments)
+        with pytest.raises(serializers.ValidationError) as exc_info:
+            action.validate_arguments(arguments)
```

Comment:
This test passes, but it's incorrect. We only create one recipe, so its slug has nothing to conflict. But, because we don't pass `old_arguments`, it treats the experiment as brand new, and compares itself to its own slug, which conflicts, and passes the test.

We'll probably want to have two tests, one for the newly-created case, and one for the existing-recipe case.